scripts/main-loop.js

```diff
@@ -74,14 +82,92 @@ define(['./packet', './loop-factory', './data-factory'], function (packet, loopF
                                                           clientName));
     };
 
-    loop.onDataCallbacks[packet.types.cmdRequest] = function (content) {
-        switch (content.opCode) {
+    loop.onDataCallbacks[packet.types.cmdRequest] = function (request) {
+
+        var data, hexString;
+
+        switch (request.opCode) {
             case operationCodes.openSession: 
-                openSessionId = content.sessionId;
-                loop.scheduleSend(packet.createCmdResponse(operationCodes.okay, content.transactionId));
+                openSessionId = request.sessionId;
+                loop.scheduleSend(packet.createCmdResponse(operationCodes.okay, request.transactionId));
+            break;
+            case operationCodes.getDeviceInfo:
+
+                hexString = "64 00 11 00 00 00 64 00 14 53 00 6f 00 6e 00 79" +
+                    "00 20 00 50 00 54 00 50 00 20 00 45 00 78 00 74" +
+                    "00 65 00 6e 00 73 00 69 00 6f 00 6e 00 73 00 00" +
+                    "00 00 00 18 00 00 00 01 10 02 10 03 10 04 10 05" +
+                    "10 06 10 07 10 08 10 09 10 0a 10 1b 10 01 92 02" +
+                    "92 05 92 07 92 09 92 0a 92 0b 92 0c 92 0d 92 01" +
+                    "98 02 98 03 98 05 98 07 00 00 00 01 c2 02 c2 03" +
+                    "c2 04 c2 05 c2 06 c2 07 c2 00 00 00 00 00 00 00" +
+                    "00 03 00 00 00 01 38 01 b3 01 b1 11 53 00 6f 00" +
+                    "6e 00 79 00 20 00 43 00 6f 00 72 00 70 00 6f 00" +
+                    "72 00 61 00 74 00 69 00 6f 00 6e 00 00 00 07 49" +
+                    "00 4c 00 43 00 45 00 2d 00 39 00 00 00 05 36 00" +
+                    "2e 00 30 00 30 00 00 00 21 30 00 30 00 30 00 30" +
+                    "00 30 00 30 00 30 00 30 00 30 00 30 00 30 00 30" +
+                    "00 30 00 30 00 30 00 30 00 33 00 32 00 38 00 32" +
+                    "00 38 00 39 00 33 00 30 00 30 00 33 00 37 00 37" +
+                    "00 37 00 39 00 38 00 32 00 00 00 0c 00 00 00 0c" +
+                    "00 00 00 01 00 00 00 0e 00 00 00 07 00 00 00 01" +
+                    "20 01 00 00 00";
+                let a9DeviceInfoPayload = dataFactory.createFromHexString(hexString);
+
+                console.log("Received Get_Device_Info", request);
+                //@simon TODO: work out why we can't send the actual data length here!
```

Comment:
I think I understand why the length doesn't work.

In line 112, `0c 00 00 00` begins "end data packet". For some reason Wireshark mangled "data packet" and "end data packet" two together. Should be relatively easy to extract it into a separate call and then get "end data packet" properties from Wireshark.

Reply:
Same story with all other blobs in this file.

Reply:
@simonmitchell Fixed in #4.